### PR TITLE
:bug: fix bug with bound_method + :sparkles: new integrations

### PR DESCRIPTION
**Pull request by jvdd**
This PR handles 

(1) :bug: a **bug** in the `bound_method` + sequence based strided rolling
- [x] check if `.loc` induces memory peak
- [x] agree on what behavior is preferred for segmentation indexing
- **Agreed behavior**: 
  - make `window_idx="begin"` default instead of `"end"`
  - sequences should be segmented into *n* segments if there are exactly *n* segments possible (e.g., window=2, stride=2 => 5 segments on sequence of length 10)
  - we remain the current behavior of the `end`-index of the segments.
- [x] extend tests

(2) :sparkles: extends **integration with other feature extraction packages**
- [x] add and test [`catch22`](https://github.com/chlubba/catch22) integration wrapper  
- [x] :x: add and test [`scikit-learn transformer`](https://scikit-learn.org/stable/data_transforms.html) wrapper
- **Decided to not do this (for now)**: have atm an ambiguas implementation -> more details in #56  

(3): :zap: faster irregular data check

(4): :fire: add kaggle TPSAPR2022 notebook to ML examples

### CODE REVIEW

File: tsflex/features/feature_collection.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def calculate(
         # determing the bounds of the series dict items and slice on them
         start, end = _determine_bounds(bound_method, list(series_dict.values()))
         series_dict = {
-            n: s[s.index.dtype.type(start) : s.index.dtype.type(end)]
+            n: s.loc[s.index.dtype.type(start) : s.index.dtype.type(end)]  # TODO: check memory efficiency of ths
```
Comment (jvdd) [inline tsflex/features/feature_collection.py:370]: Check memory efficiency of this

**Comment by jvdd** [comment]
1 tests will fail (as we need to agree what indexing is preferred).  
Until then, this PR stays a draft

File: tsflex/features/segmenter/strided_rolling.py [hunk 2]
```diff
@@ -142,17 +142,12 @@ def __init__(
...
         if not self.approve_sparsity:
-            qs = [0, 0.1, 0.5, 0.9, 1]
             for container in self.series_containers:
-                series_idx_stats = np.quantile(
-                    container.end_indexes - container.start_indexes, q=qs
-                )
-                q_str = ", ".join([f"q={q}: {v}" for q, v in zip(qs, series_idx_stats)])
                 # Warn when min != max
-                if not all(series_idx_stats == series_idx_stats[-1]):
+                if np.ptp(container.end_indexes - container.start_indexes) != 0:
```
Comment (jvdd) [inline tsflex/features/segmenter/strided_rolling.py:147]: Calculates difference between min and max.
Is significantly faster than the previous implementation (that used `np.quantile`)

In this benchmark `idx = np.arange(100_000)`
image

Comment (jonasvdd) [reply tsflex/features/segmenter/strided_rolling.py:147]: nice one! 🔥

File: tests/test_features_feature_collection.py [hunk 3]
```diff
@@ -1193,7 +1216,7 @@ def test_serialization(dummy_data):
 
     df_tmp = dummy_data["TMP"].reset_index(drop=True)
     df_eda = dummy_data["EDA"].reset_index(drop=True)
-    out = fc.calculate([df_tmp, df_eda], window_idx="end", return_df=True)
+    out = fc.calculate([df_tmp, df_eda], return_df=True)
```
Comment (jvdd) [inline tests/test_features_feature_collection.py:1219]: To make it consistent with line 1230

File: tsflex/features/feature_collection.py (hunk 1 above)
Comment (jvdd) [reply tsflex/features/feature_collection.py:370]: Runtime is the same as previous implementation :heavy_check_mark: 
image

Comment (jvdd) [reply tsflex/features/feature_collection.py:370]: Memory profiling indicates no memory peak :heavy_check_mark: 
image

**Comment by jonasvdd** [comment]
@jvdd, might also be interesting too look into: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.resample.html. Maybe we can use 
* the same naming convention
* or even re-use their underlying logic (i.e., don't reimplement the wheel)

File: examples/README.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ tsflex is a domain independent package for time series processing & feature extr
 | Climate modelling | [Ozone level detection](https://archive.ics.uci.edu/ml/datasets/Ozone%20Level%20Detection) | [example_ozone_level_detection.ipynb](https://github.com/predict-idlab/tsflex/blob/main/examples/example_ozone_level_detection.ipynb) |  
 | Household data | [Electric power consumption](https://archive.ics.uci.edu/ml/datasets/Individual+household+electric+power+consumption) | [example_power_consumption_estimation.ipynb](example_power_consumption_estimation.ipynb) |
 | Clinical data | [Sleep-EDF Database Expanded](https://physionet.org/content/sleep-edfx/1.0.0/) | [example_sleep_staging.ipynb](example_sleep_staging.ipynb) |
+| kaggle competition | [Tabular Playground Series - Apr 2022](https://www.kaggle.com/competitions/tabular-playground-series-apr-2022)| https://www.kaggle.com/code/jeroenvdd/tpsapr22-best-non-dl-model-tsflex-powershap |
```
Comment (jonasvdd) [inline examples/README.md:24]: TODO: maybe state here that the data was already segmented -> so here you can find an example on how to use `tsflex` on already segmented data?

File: tsflex/features/segmenter/strided_rolling.py [hunk 5]
```diff
@@ -112,7 +112,7 @@ def __init__(
         self.data_type = func_data_type
 
         # 0. Standardize the input
-        series_list: List[pd.Series] = to_series_list(data)
+        series_list: List[pd.Series] = to_series_list(data)  # TODO: isn't it always a list of series?
```
Comment (jonasvdd) [inline tsflex/features/segmenter/strided_rolling.py:115]: Indeed, normally it is, but when you want to solely play/develop with this class, it is convenient to still have this function (I assume this has not a large duration / memory overhead)

fair point though

**Pull request merged**